test/deployframework/helpers.go

```diff
@@ -190,3 +192,69 @@ func GetServiceAccountToken(client kubernetes.Interface, namespace, serviceAccou
 
 	return string(secret.Data["token"]), nil
 }
+
+func runCleanupScript(logger logrus.FieldLogger, namespace, outputPath, scriptPath string) error {
+	var errArr []string
+	envVarArr, err := createResourceDirs(namespace, outputPath)
+	if err != nil {
+		errArr = append(errArr, fmt.Sprintf("failed to create the resource output directories: %v", err))
+	}
+
+	cleanupCmd := exec.Command(scriptPath)
+	cleanupStdout, err := cleanupCmd.StdoutPipe()
+	if err != nil {
+		errArr = append(errArr, fmt.Sprintf("failed to create a pipe from command output to stdout: %v", err))
+	}
+
+	go func() {
+		scanner := bufio.NewScanner(cleanupStdout)
+		for scanner.Scan() {
+			line := scanner.Text()
+			logger.Infof(line)
+		}
+		if err := scanner.Err(); err != nil {
+			errArr = append(errArr, fmt.Sprintf("error reading output from command: %v", err))
+		}
+		return
+	}()
+
+	cleanupCmd.Env = append(os.Environ(), envVarArr...)
+	err = cleanupCmd.Run()
+	if err != nil {
+		errArr = append(errArr, fmt.Sprintf("error running the cleanup script: %v", err))
+	}
+
+	if len(errArr) != 0 {
+		return fmt.Errorf(strings.Join(errArr, "\n"))
+	}
+
+	return nil
+}
+
+func cleanupPID(logger logrus.FieldLogger, pids ...int) error {
+	var errArr []string
+
+	logger.WithFields(logrus.Fields{
+		"cleanup": pids,
+	}).Infof("Killing the local process IDs")
+
+	for _, pid := range pids {
+		p, err := os.FindProcess(pid)
+		if err != nil {
+			errArr = append(errArr, fmt.Sprintf("failed to find the %d pid: %v", pid, err))
+		}
+
+		err = p.Kill()
```

Comment:
Does the original script `kill -9` or just `kill`? If the latter, we should use `p.Signal(signal.Interrupt)` so it's graceful.

Reply:
It was just a regular `kill`:
```bash
    if [ "$DEPLOY_METERING_OPERATOR_LOCAL" == "true" ]; then
        echo "Stopping local metering-operator"
        [ -s "$METERING_OPERATOR_PID_FILE" ] && kill "$(cat "$METERING_OPERATOR_PID_FILE")" || true
        rm -f "$METERING_OPERATOR_PID_FILE"
        docker rm -f "$METERING_OPERATOR_CONTAINER_NAME" || true
    fi
```

Reply:
Then we should be using `Signal` not `Kill`